client/lib/managementClient.ts

```diff
@@ -65,26 +68,48 @@ export interface EventHubPartitionRuntimeInformation {
   type: "com.microsoft:partition";
 }
 
+export interface ManagementClientOptions {
+  address?: string;
+  audience?: string;
+}
+
 /**
  * @class ManagementClient
  * Descibes the EventHubs Management Client that talks
  * to the $management endpoint over AMQP connection.
  */
-export class ManagementClient {
+export class ManagementClient extends ClientEntity {
 
   readonly managementLock: string = `${Constants.managementRequestKey}-${uuid()}`;
+  /**
+   * @property {string} entityPath - The name/path of the entity (hub name) for which the management
+   * request needs to be made.
+   */
+  entityPath: string;
+  /**
+   * @property {string} replyTo The reply to Guid for the management client.
+   */
+  replyTo: string = uuid();
   /**
    * $management sender, receiver on the same session.
+   * @private
    */
   private _mgmtReqResLink?: RequestResponseLink;
 
   /**
    * @constructor
    * Instantiates the management client.
-   * @param entityPath - The name/path of the entity (hub name) for which the management request needs to be made.
-   */
-  constructor(public entityPath: string) {
-    this.entityPath = entityPath;
+   * @param {BaseConnectionContext} context The connection context.
+   * @param {string} [address] The address for the management endpoint. For IotHub it will be
+   * `/messages/events/$management`.
+   */
+  constructor(context: ConnectionContext, options?: ManagementClientOptions) {
+    super(context, {
+      address: options && options.address ? options.address : Constants.management,
+      audience: options && options.audience ? options.audience : context.config.endpoint
```

Comment:
Consider setting up your own options structure with what you think it should be then merging in the passed options?  This way caller could send down options you haven't thought about here? Just a thought.